gpustack/worker/model_file_manager.py

```diff
@@ -95,10 +102,112 @@ async def _handle_deletion(self, model_file: ModelFile):
+                    self._config.cache_dir,
+                    SourceEnum.HUGGING_FACE,
+                    model_file.huggingface_repo_id,
+                    model_file.huggingface_filename,
+                )
+                path_obj = Path(str(path))
+                filename_pattern = path_obj.name
+                local_dir = path_obj.parent
+                download_paths = get_local_download_paths(local_dir, filename_pattern)
+                cache_dir = download_paths.lock_path.parent
+
+                # Get actual filename by pattern
+                for cache_file in await asyncio.to_thread(
+                    glob.glob, str(cache_dir / filename_pattern) + "*"
+                ):
+                    # cut off the path and useless extension
+                    filename = cache_file.rsplit("/", 1)[-1]
+                    filename = filename.rsplit(".", 1)[0]
+                    break
+
+                metadata = await self.get_hf_file_metadata(model_file, filename)
+
+                # Collect lock files and incomplete files
+                paths_to_delete.add(str(cache_dir / (filename + ".lock")))
+                for item_path_str in await asyncio.to_thread(
+                    glob.glob, str(cache_dir / f"*.{metadata.etag}.incomplete")
+                ):
+                    paths_to_delete.add(item_path_str)
+
+            elif model_file.source == SourceEnum.MODEL_SCOPE:
+                path = os.path.join(
+                    self._config.cache_dir,
+                    SourceEnum.MODEL_SCOPE,
+                    model_file.model_scope_model_id,
+                    model_file.model_scope_file_path,
+                )
+                path_obj = Path(str(path))
+                filename_pattern = path_obj.name
+                local_dir = path_obj.parent
+                for delete_file in await asyncio.to_thread(
+                    glob.glob,
+                    str(local_dir / f"{TEMPORARY_FOLDER_NAME}/{filename_pattern}"),
+                ):
+                    paths_to_delete.add(delete_file)
+
+        except Exception as e:
+            logger.error(
+                f"Error deleting incomplete Download files for "
+                f"file '{filename}': {e}"
+            )
+
+        return paths_to_delete
+
+    async def _delete_incomplete_model_files(self, model_file: ModelFile):
+        if model_file.state != ModelFileStateEnum.DOWNLOADING:
+            return
+
+        # If the model is not in GGUF format, directly delete the entire directory
+        if not model_file.model_scope_file_path and not model_file.huggingface_filename:
+            path = os.path.join(
+                self._config.cache_dir,
+                model_file.source,
+                model_file.model_scope_model_id or model_file.huggingface_repo_id,
+            )
+            paths_to_delete = [path]
+        else:
+            paths_to_delete = await self._get_incomplete_model_files(model_file)
```

Comment:
Can we put this logic in _get_incomplete_model_files so that it's self-contained?